docs/content/Deployment/Platform-Serverless-AWS.md

```diff
@@ -0,0 +1,199 @@
+Framework][link-sls] on [AWS][link-aws].
+
+<!-- prettier-ignore-start -->
+[[warning |]]
+| This is an example of a production-ready deployment, but real-world
+| deployments can vary significantly depending on desired performance
+| and scale.
+<!-- prettier-ignore-end -->
+
+## Prerequisites
+
+- An AWS account
+- An [Elasticache Redis][aws-redis] cluster URL
+- A separate Cube Store or external database
+- Node.js 10+
+- Serverless Framework
+
+## Configuration
+
+Create a Serverless Framework project by creating a `serverless.yml`. A
+production-ready stack would at minimum consist of:
+
+- One or more Cube.js API instance
+- A Cube.js Refresh Worker
+
+The `serverless.yml` for an example project is provided below:
+
+```yaml
+service: hello-cube-sls
+
+provider:
+  name: aws
+  runtime: nodejs12.x
+  iamRoleStatements:
+    - Effect: 'Allow'
+      Action:
+        - 'sns:*'
+      Resource: '*'
+  environment:
+    CUBEJS_DB_HOST: <YOUR_DB_HOST_HERE>
+    CUBEJS_DB_NAME: <YOUR_DB_NAME_HERE>
+    CUBEJS_DB_USER: <YOUR_DB_USER_HERE>
+    CUBEJS_DB_PASS: <YOUR_DB_PASS_HERE>
+    CUBEJS_DB_PORT: <YOUR_DB_PORT_HERE>
+    CUBEJS_REDIS_URL: <YOUR_REDIS_URL_HERE>
+    CUBEJS_DB_TYPE: <YOUR_DB_TYPE_HERE>
+    CUBEJS_API_SECRET: <YOUR_API_SECRET_HERE>
+    CUBEJS_APP: '${self:service.name}-${self:provider.stage}'
+    NODE_ENV: production
+    AWS_ACCOUNT_ID:
+      Fn::Join:
+        - ''
+        - - Ref: 'AWS::AccountId'
+
+functions:
+  cubejs:
+    handler: index.api
+    timeout: 30
+    events:
+      - http:
+          path: /
+          method: GET
+      - http:
+          path: /{proxy+}
+          method: ANY
+          cors:
+            origin: '*'
+            headers:
+              - Content-Type
+              - Authorization
+              - X-Request-Id
+              - X-Amz-Date
+              - X-Amz-Security-Token
+              - X-Api-Key
+  cubejsProcess:
+    handler: index.process
+    timeout: 630
+    events:
+      - sns: '${self:service.name}-${self:provider.stage}-process'
+
+plugins:
+  - serverless-express
+```
+
+### Refresh Worker
+
+<!-- prettier-ignore-start -->
+[[warning |]]
+| Running a refresh worker using Serverless requires a _slightly_ different
+| setup than Docker. You must continuously call the endpoint once every 60
+| seconds to keep the pre-aggregation creation queries in the queue. Failing to
+| do this will prevent pre-aggregations from being built.
+<!-- prettier-ignore-end -->
+
+To begin the scheduled refresh, we must call the
+[`/v1/run-scheduled-refresh`][ref-restapi-sched-refresh] endpoint. The endpoint
+will return `{ "finished": false }` whilst the pre-aggregations are being built;
+once they are successfully built, the response will change to:
+
+```json
+{
+  "finished": true
+}
+```
+
+### Cube Store
+
+Unfortunately Cube Store cannot currently be run using serverless platforms; we
+strongly recommend using a PaaS such as [AWS EC2][aws-ec2], [AWS ECS][aws-ecs]
```

Comment:
@hassankhan Should we skip naming any services? We haven't run cube store with a production load on any of these services therefore I'm not sure we can strongly recommend one of them. We can just say that Cube Store should be deployed differently on services like AWS EC2, ECS, etc.

Reply:
Good point, rewrote that bit and added a link back to Caching > Running in Production.